autoload/asyncomplete/sources/tmuxcomplete.vim

```diff
@@ -0,0 +1,38 @@
+function! asyncomplete#sources#tmuxcomplete#register(opts)
+    let l:opts = extend(extend({}, a:opts), {
+                \ 'name':      'tmuxcomplete',
+                \ 'whitelist': ['*'],
+                \ 'completor': function('asyncomplete#sources#tmuxcomplete#completor'),
+                \ })
+    call asyncomplete#register_source(l:opts)
+endfunction
+
+function! asyncomplete#sources#tmuxcomplete#completor(opt, ctx)
+    " taken from asyncomplete-buffer
+    let l:kw = matchstr(a:ctx['typed'], '\w\+$')
+    let a:opt['kwlen'] = len(l:kw)
```

Comment:
You should treat `a:opt` and `a:ctx` as immutable . You can create your own dict instead